### PR TITLE
WIP: Add like a dapp lb

### PR DESCRIPTION
**Pull request by andrewkryshtal**

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -745,4 +742,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: b543d06d281fa28215d885e327619db02fdf3071
 
-COCOAPODS: 1.11.2
+COCOAPODS: 1.10.2
```
Comment (unixvb) [inline ios/Podfile.lock:745]: update your local cocoapods version

File: src/utils/number.util.ts [hunk 2]
```diff
@@ -6,9 +6,7 @@ export const formatAssetAmount = (amount: BigNumber, decimalPlace = 6) => {
     return '';
   }
 
-  return numberWithSpaces(
-    amount.decimalPlaces(amount.abs().lt(1000) ? decimalPlace : 2, BigNumber.ROUND_DOWN).toFixed()
-  );
+  return numberWithSpaces(amount.decimalPlaces(amount.abs().lt(1000) ? decimalPlace : 2).toFixed());
```
Comment (unixvb) [inline src/utils/number.util.ts:9]: you have lost `BigNumber.ROUND_DOWN` argument

File: src/screens/lb-dapp/lb-dapp.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        <ButtonLargePrimary
+          style={styles.buttonStyles}
+          title={'REMOVE'}
+          iconName={IconNameEnum.MinusIcon}
+          onPress={onRemoveLiquidityHandler}
+        />
+        <Divider size={formatSize(16)} />
+        <ButtonLargePrimary
+          style={styles.buttonStyles}
+          title={'ADD'}
```
Comment (unixvb) [inline src/screens/lb-dapp/lb-dapp.tsx:67]: title="ADD"   
title="REMOVE"

File: src/screens/lb-dapp/lb-dapp.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  return (
+    <ScreenContainer>
+      <InsetSubstitute type="top" />
+      <View style={styles.lineDivider} />
+      <Divider size={formatSize(8)} />
+      <View style={styles.lbCoinContainer}>
+        <LpTokenIcon firstTokenIcon={IconNameEnum.LbTokenIcon} secondTokenIcon={IconNameEnum.LbTokenIcon} />
+        <Text style={styles.lbCoinText}>XTZ/tzBTC</Text>
+        <Divider size={formatSize(8)} />
+        <TouchableIcon name={IconNameEnum.RefreshIcon} size={formatSize(24)} onPress={onRefreshPressHandler} />
```
Comment (unixvb) [inline src/screens/lb-dapp/lb-dapp.tsx:43]: This's not a refresh button, it should change the direction of chart (xtz/btc > btc/xtz)
remove it for now

**Review by unixvb — APPROVED** [review]

File: src/modals/remove-liquidity-modal/remove-liquidity-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+  const aTokenPool = storage.xtzPool;
+  const bTokenPool = storage.tokenPool;
+  const lpTotalSupply = storage.lqtTotal;
+
+  const { publicKeyHash } = useSelectedAccountSelector();
+  const styles = useRemoveLiquidityModalStyles();
+  const assetsList = useAssetsListSelector();
+  const tokenA = useTezosTokenSelector();
+
+  const lpList = assetsList.filter(token => token.address === 'KT1AafHA1C1vk959wvHWBispY9Y2f3fxBUUo');
```
Comment (unixvb) [inline src/modals/remove-liquidity-modal/remove-liquidity-modal.tsx:54]: create const with this address

File: src/modals/remove-liquidity-modal/remove-liquidity-modal.tsx [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+  const tokenB = assetsList.filter(token => token.name === 'tzBTC')[0];
```
Comment (unixvb) [inline src/modals/remove-liquidity-modal/remove-liquidity-modal.tsx:55]: use `find` method instead and search by address&id

File: src/navigator/main-stack.tsx [hunk 7]
```diff
@@ -151,6 +152,13 @@ export const MainStackScreen = () => {
                 component={DApps}
                 options={{ animationEnabled: false, headerShown: false }}
               />
+              <MainStack.Screen
+                name={ScreensEnum.LbDapp}
+                component={LbDapp}
+                options={{
+                  headerShown: false
+                }}
```
Comment (unixvb) [inline src/navigator/main-stack.tsx:160]: could this be inlined (as in `ScreensEnum.DApps` `options`)?

File: src/op-params/liquidity-baking/remove-liquidity.op-params.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
+export const removeLiquidityOpParams = {
```
Comment (unixvb) [inline src/op-params/liquidity-baking/remove-liquidity.op-params.ts:1]: delete this

File: src/screens/d-apps/d-apps.tsx [hunk 9]
```diff
@@ -36,6 +39,19 @@ export const DApps = () => {
...
       <SearchInput placeholder="Search token" onChangeText={setSearchQuery} />
       <Divider size={formatSize(20)} />
+      <Text style={styles.text}>Integrated</Text>
+      <Divider size={formatSize(20)} />
+      <View style={styles.dappBlockWrapper}>
+        <IntegratedDApp
+          screenName={ScreensEnum.LiquidityBakingDapp}
+          iconName={IconNameEnum.LbDappIcon}
+          title={'Liquidity Baking'}
+          description={'Create XTZ/tzBTC & earn XTZ'}
```
Comment (unixvb) [inline src/screens/d-apps/d-apps.tsx:49]: the same comment as it was before

File: src/screens/lb-dapp/lb-dapp.tsx [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+import { useLbDappStyles } from './lb-dapp.styles';
+
+export const LbDapp = () => {
+  const { navigate } = useNavigation();
+  const styles = useLbDappStyles();
+
+  // const onRefreshPressHandler = () => {
+  //   console.log('refresh pressed');
+  // };
+
```
Comment (unixvb) [inline src/screens/lb-dapp/lb-dapp.tsx:25]: ?

File: src/screens/lb-dapp/lb-dapp.tsx [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+      <ButtonsContainer>
+        <ButtonLargePrimary
+          style={styles.buttonStyles}
+          title="REMOVE"
+          iconName={IconNameEnum.MinusIcon}
+          onPress={onRemoveLiquidityHandler}
+        />
+        <Divider size={formatSize(16)} />
+        <ButtonLargePrimary
+          style={styles.buttonStyles}
```
Comment (unixvb) [inline src/screens/lb-dapp/lb-dapp.tsx:66]: Do not give the ability to pass `styles` to the shared components (they are already styled).  
Take a look at how your problem was solved at Home screen (SEND/RECEIVE/BUY buttons)

File: src/utils/number.util.spec.ts [hunk 12]
```diff
@@ -26,11 +26,11 @@ describe('formatAssetAmount', () => {
   });
 
   it('should format positive bignumber more than 1000 value and return string with 2 decimals using rounding mode up', () => {
-    expect(formatAssetAmount(bigNumberMoreThanThousand)).toEqual('10 000.25');
+    expect(formatAssetAmount(bigNumberMoreThanThousand)).toEqual('10 000.26');
   });
 
   it('should format positive bignumber less than 1000 value and return string with 1 decimal using rounding mode up', () => {
-    expect(formatAssetAmount(new BigNumber(100.255133), 1)).toEqual('100.2');
+    expect(formatAssetAmount(new BigNumber(100.255133), 1)).toEqual('100.3');
```
Comment (unixvb) [inline src/utils/number.util.spec.ts:33]: run units test, does this work?

**Review by unixvb — CHANGES_REQUESTED** [review]

**Review by LendiHop — APPROVED** [review]

File: src/screens/lb-dapp/lb-dapp.tsx (hunk 11 above)
Comment (andrewkryshtal) [reply src/screens/lb-dapp/lb-dapp.tsx]: there is no width fix for buttons in home screen, to align buttons we should add custom styles to buttons to align them correctly, or make a prop with custom width for each button

Comment (unixvb) [reply src/screens/lb-dapp/lb-dapp.tsx]: just wrap each button with `buttonContainer` `View` that will set the with for local specific case

**Review by unixvb — APPROVED** [review]

**Pull request merged**